### ISSUE TITLE
Issue with DataFrame.rename and the columns named argument

### ISSUE THREAD

**Issue by gerardsyd** [microsoft/pylance-release#71]
## Environment data

-   Language Server version: 2020.6.1
-   OS and version: Windows 10.1903
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.7.7
-   Pandas version: 1.0.5

## Expected behaviour

When using df.rename() and passing columns as a named argument, no error is raised as this is per the documentation:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.rename.html 

## Actual behaviour

Pylance shows an error saying no parameter named "columns". I've recreated the issue in the code below. It seems to have something to do with using groupby which leads Pylance to believe the DataFrame is a Series object

![image](https://user-images.githubusercontent.com/62287799/86875686-6d023280-c126-11ea-954a-29e09dd7f736.png)

## Code Snippet / Additional information

```python
import pandas as pd


df = pd.DataFrame({'col1': ['a', 'b', 'b', 'd'], 'Data': [1, 2, 3, 4]})
print(df)
df = df.groupby('col1').last().reset_index()
print(df)

df.rename(columns={'Data': 'Test'}, inplace=True)
print(df)
```

**Comment by erictraut** [comment]
Thanks for the bug report.

@gramster, this looks like a bug in the pandas type stub?

As for the error, I've changed the diagnostics related to argument/parameter matching to be suppressed when type checking mode is disabled. This change will be in the next version of Pylance.

**Comment by gramster** [comment]
Yes, that stub is an older one from when a single scaler ‘by’ argument
would return a SeriesGroupBy. That is no longer true and the overload
should be removed. Will fix it tomorrow.

**Comment by tamuhey** [comment]
Pylance also infer the below code incorrectly:

```python
import pandas as pd

df = pd.DataFrame({"a": [1, 2, 3]})
print(df.fillna("")) # Error: No overloads for "df.fillna("")" match parameters
```

**Comment by CatNofishing** [comment]
Some functions have no intelligent suggestions in Pandas.  For example  : 
```
import pandas as pd
frame=pd.DataFrame()
```
![image](https://user-images.githubusercontent.com/33288114/86989981-2f58e480-c1ce-11ea-9b96-ef9b6e25747c.png)

**Comment by erictraut** [comment]
@gramster, this looks like a bug in the pandas type stub. The `__init__.pyi` stub is importing `DataFrame`, but it's not re-exporting it in the top-level namespace for the pandas module.

To re-export a symbol from a type stub, you need to use the "as" syntax (as per PEP 484). Something like this...
```
from .core import DataFrame as DataFrame
```

**Comment by jakebailey** [comment]
That's #13.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.7.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202070-9-july-2020
